hs_app_timeseries/models.py

```diff
@@ -950,211 +1106,462 @@ def delete_all_elements(self):
-                    utils.replace_resource_file_on_irods(temp_sqlite_file, sqlite_file_to_update,
-                                                         user)
-                    self.is_dirty = False
-                    self.save()
-                    log.info("SQLite file update was successful.")
-            except sqlite3.Error as ex:
-                sqlite_err_msg = str(ex.args[0])
-                log.error("Failed to update SQLite file. Error:{}".format(sqlite_err_msg))
-                raise Exception(sqlite_err_msg)
-            except Exception as ex:
-                log.exception("Failed to update SQLite file. Error:{}".format(ex.message))
-                raise ex
-            finally:
-                if os.path.exists(temp_sqlite_file):
-                    shutil.rmtree(os.path.dirname(temp_sqlite_file))
+    def update_datasets_table_insert(self, con, cur):
+        # insert record to Datasets table - first delete any existing records
+        # used for updating a sqlite file that is blank (case of CSV upload)
 
-    def populate_blank_sqlite_file(self, temp_sqlite_file, user):
-        """
-        writes data to a blank sqlite file. This function is executed only in case
-        of CSV file upload and executed only once when updating the sqlite file for the first time.
-        :param temp_sqlite_file: this the sqlite file copied from irods to a temp location on
-         Django.
-        :param user: current user (must have edit permission on resource) who is updating the
-        sqlite file to sync metadata changes in django database.
-        :return:
-        """
-        log = logging.getLogger()
-        if not self.resource.has_csv_file:
-            msg = "Resource needs to have a CSV file before a blank SQLite file can be updated"
-            log.exception(msg)
-            raise Exception(msg)
+        cur.execute("DELETE FROM Datasets")
+        con.commit()
+        insert_sql = "INSERT INTO Datasets (DatasetID, DatasetUUID, DatasetTypeCV, " \
+                     "DatasetCode, DatasetTitle, DatasetAbstract) VALUES(?,?,?,?,?,?)"
 
-        # update resource specific metadata element series ids with generated GUID
-        self._update_metadata_element_series_ids_with_guids()
+        if isinstance(self, TimeSeriesMetaData):
+            ds_title = self.title.value
+            ds_abstract = self.description.abstract
+            ds_code = self.resource.short_id
+        else:
+            # self must be TimeSeriesFileMetaData
+            ds_title = self.logical_file.dataset_name
+            ds_abstract = self.abstract
+            ds_code = self.logical_file.resource.short_id
 
-        blank_sqlite_file = utils.get_resource_files_by_extension(self.resource, ".sqlite")[0]
-        csv_file = utils.get_resource_files_by_extension(self.resource, ".csv")[0]
+        cur.execute(insert_sql, (1, uuid4().hex, 'Multi-time series', ds_code, ds_title,
+                                 ds_abstract), )
```

Comment:
I assume DatasetID is not defined as a primary key, otherwise, uuid4().hex may present uniqueness issue.

Reply:
As far as I know sqlite file doesn't have that kind of constraint. However, I am wondering how come we are using uuid4().hex for resource id?

Reply:
In fact DataSetID is a primary key.

Reply:
@pkdash if DatasetID is a primary key, perhaps it is better not to use uuid4().hex for it since theoretically uuid4() cannot guarantee uniqueness, but use AutoField instead to leave id generation to Django? resource_id is not a primary key, although resource id duplication would be very bad, statistically, the chance for uuid duplication is very small and is negligible. In any case, I'll defer this to your judgement whether you want to keep using uuid4 or AutoField for DataSetID.

Reply:
@hyi The record that is being created here is not in the Django database but in the Sqlite file -so can't use AutoField.